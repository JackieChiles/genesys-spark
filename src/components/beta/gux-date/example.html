<h1>gux-date-beta</h1>

<h2>Dates</h2>

<h3>Short</h3>
<gux-date-beta id="date-1" format="short"></gux-date-beta>

<h3>Medium</h3>
<gux-date-beta id="date-2" format="medium"></gux-date-beta>

<h3>Long</h3>
<gux-date-beta id="date-3" format="long"></gux-date-beta>

<h3>Full</h3>
<gux-date-beta id="date-4" format="full"></gux-date-beta>

<style
  onload="(function() {
    const date1 = document.querySelector('#date-1');
    const date2 = document.querySelector('#date-2');
    const date3 = document.querySelector('#date-3');
    const date4 = document.querySelector('#date-4');

    const date = new Date();

    date1.date = date;
    date2.date = date;
    date3.date = date;
    date4.date = date;
  })()"
>
  .not-used {
    -custom-noop: noop;
  }
</style>
