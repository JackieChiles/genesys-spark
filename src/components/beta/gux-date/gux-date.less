@import (reference) '../../../style/color.less';
@import (reference) '../../../style/spacing.less';

.gux-date {
    display: flex;
    padding: @spacing-xs @spacing-xs;
    color: @gux-black-10;
}