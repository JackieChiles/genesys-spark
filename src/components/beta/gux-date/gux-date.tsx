import { Component, h, Prop, Element } from '@stencil/core';
import { trackComponent } from '../../../usage-tracking';
import { DateTimeFormatter, GuxDateTimeFormat } from '../../../i18n/DateTimeFormatter';
import { getDesiredLocale } from '../../../i18n/index';

@Component({
  styleUrl: 'gux-date.less',
  tag: 'gux-date-beta',
  shadow: true
})
export class GuxDate {
  private formatter : DateTimeFormatter;

  /**
   * Reference to the host element.
   */
  @Element()
  root: HTMLElement;

  /**
   * The Date object to format 
   */
  @Prop()
  date: Date = new Date();

  /**
   * Format option type 
   */
  @Prop()
  format: GuxDateTimeFormat = 'full';

  componentWillLoad(): void {
    trackComponent(this.root);
    this.formatter = new DateTimeFormatter(getDesiredLocale(this.root));
  }

  render() {
    return (<span>{ this.formatter.formatDate(this.date, this.format) }</span>);
  }
}
