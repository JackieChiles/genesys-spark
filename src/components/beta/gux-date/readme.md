# gux-date-beta



<!-- Auto Generated Below -->


## Properties

| Property   | Attribute  | Description                                         | Type                                      | Default                    |
| ---------- | ---------- | --------------------------------------------------- | ----------------------------------------- | -------------------------- |
| `datetime` | `datetime` | The ISO string representation of the date to format | `string`                                  | `new Date().toISOString()` |
| `format`   | `format`   | Format option type                                  | `"full" \| "long" \| "medium" \| "short"` | `'short'`                  |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
