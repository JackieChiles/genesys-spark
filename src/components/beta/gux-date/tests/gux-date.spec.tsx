import { h } from '@stencil/core';
import { newSpecPage } from '@stencil/core/testing';
import { GuxDate } from '../gux-date';

const components = [GuxDate];
const language = 'en';
const date = new Date(2022, 6, 7, 9, 35, 30, 100);

describe('#render different datetime formats eg(short,medium,full,long)', () => {
  [
    () => (<gux-date-beta date={date} format="short"></gux-date-beta>),
    () => (<gux-date-beta date={date} format="medium"></gux-date-beta>),
    () => (<gux-date-beta date={date} format="full"></gux-date-beta>),
    () => (<gux-date-beta date={date} format="long"></gux-date-beta>)
  ].forEach((template, index) => {
    it(`should render component as expected (${index + 1})`, async () => {
      const page = await newSpecPage({ components, template, language });

      expect(page.root).toMatchSnapshot();
    });
  });
});
