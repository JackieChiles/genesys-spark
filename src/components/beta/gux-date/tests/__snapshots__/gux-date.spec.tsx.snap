// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render different datetime formats eg(short,medium,full,long) should render component as expected (1) 1`] = `
<gux-date-beta>
  <mock:shadow-root>
    <span>
      7/7/22
    </span>
  </mock:shadow-root>
</gux-date-beta>
`;

exports[`#render different datetime formats eg(short,medium,full,long) should render component as expected (2) 1`] = `
<gux-date-beta>
  <mock:shadow-root>
    <span>
      Jul 7, 2022
    </span>
  </mock:shadow-root>
</gux-date-beta>
`;

exports[`#render different datetime formats eg(short,medium,full,long) should render component as expected (3) 1`] = `
<gux-date-beta>
  <mock:shadow-root>
    <span>
      Thursday, July 7, 2022
    </span>
  </mock:shadow-root>
</gux-date-beta>
`;

exports[`#render different datetime formats eg(short,medium,full,long) should render component as expected (4) 1`] = `
<gux-date-beta>
  <mock:shadow-root>
    <span>
      July 7, 2022
    </span>
  </mock:shadow-root>
</gux-date-beta>
`;
